core/genesis_write.go

```diff
@@ -522,9 +524,14 @@ func GenesisToBlock(g *types.Genesis, tmpDir string, logger log.Logger) (*types.
 		}
 	}
 
-	var requests []*types.Request // TODO(racytech): revisit this after merge, make sure everythin is correct
+	var requests types.Requests
 	if g.Config != nil && g.Config.IsPrague(g.Timestamp) {
-		requests = []*types.Request{}
+		requests = types.Requests{}
+		if g.RequestsRoot != nil {
```

Comment:
I'd actually remove this functionally of supporting custom `RequestsRoot` in genesis. If doesn't work a.t.m. anyway because `types.NewBlock` at the end of this function overrides `head.RequestsRoot`.